src/config_store.c

```diff
@@ -92,9 +108,53 @@ void ConfigStore_Init(ConfigStore *p)
     p->_fd = -1;
 }
 
+/*To delete the leftover tmp files on the device startup*/
+void DeleteFileHelper(struct dirent *file, char *filePath)
+{
+    char *ptr, *fileName;
+    fileName = file->d_name;
+
+    //rindex() is a string handling function that returns a pointer to the last occurrence
+    //of character c in string s, or a NULL pointer if c does not occur in the string.
+    ptr = rindex(fileName, '.');
+
+    //Check for filename extensions
+    if ((ptr != NULL) && ((strcmp(ptr, ".tmp") == 0)))
```

Comment:
Can you use something like strncmp instead of strcmp?

Reply:
It also looks like you have an additional unnecessary set of parentheses around the strcmp.

Reply:
strcmp is replaced with strncmp in the code and unnecessary parentheses are removed.